src/mcts/search.cc

```diff
@@ -718,17 +725,29 @@ std::pair<Node*, bool> Search::PickNodeToExtend(Node* node,
   }
 }
 
-std::pair<Move, Move> Search::GetBestMove() const {
+std::tuple<Move, Move, int> Search::GetBestMove() const {
   SharedMutex::SharedLock lock(nodes_mutex_);
   Mutex::Lock counters_lock(counters_mutex_);
-  return GetBestMoveInternal();
-}
 
-std::pair<Move, Move> Search::GetBestMoveInternal() const
-    REQUIRES_SHARED(nodes_mutex_) REQUIRES_SHARED(counters_mutex_) {
-  if (responded_bestmove_) return best_move_;
-  if (!root_node_->HasChildren()) return {};
+  auto best = GetBestMoveInternal();
+
+  // guard for the case where we aren't ever resigning
+  if (kResignPercentage == 0) { 
+    return std::make_tuple(best.first, best.second, 0);
+  }
 
+  // if resigning is enabled, we check if we should resign.
+  auto win_percent = GetBestNodeInternal()->GetQ(0, 0) + 1;
```

Comment:
>GetQ returns values from -1 to 1. Therefore percent will be 0% .. 200% in this case. Should be divided by 2.


This was intentional. From my understanding, it only matters when it's negative, because that's what the bot that would resign would see. If GetQ is *positive*, the current side has an advantage, and can't trigger the other player's resignation. 



> Also win_percent doesn't really contain percents, it's in scale of 0..2 at that point, not 0..200, so the name is incorrect.



Good catch.